packages/hosted-form-v2/src/hosted-form.ts

```diff
@@ -97,6 +105,38 @@ export default class HostedForm implements HostedFormInterface {
         return this._getFirstField().submitManualOrderForm(payload.data);
     }
 
+    async submit(
+        payload: OrderPaymentRequestBody,
+        paymentIntegrationService: PaymentIntegrationService,
+        payloadTransformer: HostedFormOrderDataTransformer,
+        additionalActionData?: PaymentAdditionalAction,
+    ): Promise<HostedInputSubmitSuccessEvent> {
+        try {
+            const response = await this._getFirstField().submitForm(
+                this._fields.map((field) => field.getType()),
+                payloadTransformer.transform(payload, additionalActionData),
+            );
+
+            return response;
+        } catch (error) {
+            let additionalAction: PaymentAdditionalAction;
+
+            if (error instanceof Error || typeof error === 'string') {
+                additionalAction = await paymentIntegrationService.handlePaymentHumanVerification(
+                    error,
+                );
+            } else {
+                // Handle cases where error is not an instance of Error or string
+                throw new Error('Unexpected error type');
+            }
+
+            return await this._getFirstField().submitForm(
+                this._fields.map((field) => field.getType()),
+                payloadTransformer.transform(payload, additionalAction),
+            );
```

Comment:
```suggestion
            const response = await this._getFirstField().submitForm(
                this._fields.map((field) => field.getType()),
                payloadTransformer.transform(payload, additionalActionData),
            );

            return response;
```

Reply:
shouldn't be `additionalAction` instead proposed `additionalActionData` ?

Reply:
Yes, my bad, just copied from `try` block.